### PR TITLE
Bug fix: Fix bugs with workflow-compile/legacy

### PR DESCRIPTION
**Pull request by eggplantzzz**
There were a couple of oversights while updating the compilation packages and trying to maintain backwards compatibility with the version of workflow-compile exposed in `@truffle/core`. This PR creates a couple of adapters for `compile-vyper` and `external-compile` in `workflow-compile/legacy`, edits a method name that wasn't updated, and sends data to the events system in the proper format.

This issue was surfaced in https://github.com/trufflesuite/truffle/issues/3371 by @cgewecke. These changes seem to resolve the issues experienced with `solidity-coverage` as I tested them locally (with a bit of doctoring up of the plugin code).

### CODE REVIEW

File: packages/workflow-compile/legacy/index.js [hunk 1]
```diff
@@ -92,18 +92,53 @@ const Contracts = {
...
-        const compileFunc = multiPromisify(
-          config.all === true || config.compileAll === true
-            ? compile.all
-            : compile.necessary
-        );
-
-        let [contracts, output, compilerUsed] = await compileFunc(config);
+        config.compilersInfo = [];
+        let contracts, output, compilerUsed;
+        if (compiler === "solc") {
```
Comment (gnidan) [inline packages/workflow-compile/legacy/index.js:97]: What happens if `compiler` isn't passed?

File: packages/workflow-compile/legacy/index.js [hunk 2]
```diff
@@ -92,18 +92,53 @@ const Contracts = {
...
+            }
+          );
+          contracts = result.contracts;
+          output = result.output;
+          if (compilations[0] && compilations[0].compiler) {
+            compilerUsed = {
+              name: compilations[0].compiler.name,
+              version: compilations[0].compiler.version
+            };
+          }
```
Comment (gnidan) [inline packages/workflow-compile/legacy/index.js:137]: Should we use the `NewToLegacy` shims for these?

Comment (eggplantzzz) [reply packages/workflow-compile/legacy/index.js:137]: I don't think those shims get us what we want, do they? The only shims in there are the shims for a contract and the bytecode.

Comment (gnidan) [reply packages/workflow-compile/legacy/index.js:137]: Don't we need to shim the bytecode?

Comment (eggplantzzz) [reply packages/workflow-compile/legacy/index.js:137]: Oh, I suppose you are right about that. Doh!

**Review by gnidan — APPROVED** [review]
Looks good!

**Comment by gnidan** [comment]
Merging this so I can have the latest to rebase #2713

**Pull request merged**